include/ocpp/common/message_queue.hpp

```diff
@@ -154,6 +161,16 @@ template <typename M> class MessageQueue {
         EVLOG_debug << "Notified message queue worker";
     }
 
+    // Unlike the public resume(), this doesn't schedule anything - it just does the actual resumption
+    // Does not lock this->message_mutex, make sure to do that before calling
```

Comment:
Suggestion: Change to recursive mutex and just lock here as well if that is needed.

Reply:
Done